Algorithm.Framework/Portfolio/AccumulativeInsightPortfolioConstructionModel.cs

```diff
@@ -0,0 +1,243 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using QuantConnect.Algorithm.Framework.Alphas;
+using QuantConnect.Data.UniverseSelection;
+
+namespace QuantConnect.Algorithm.Framework.Portfolio
+{
+    /// <summary>
+    /// Provides an implementation of <see cref="IPortfolioConstructionModel"/> that allocates percent of account
+    /// to each insight, defaulting to 3%.
+    /// For insights of direction <see cref="InsightDirection.Up"/>, long targets are returned and
+    /// for insights of direction <see cref="InsightDirection.Down"/>, short targets are returned.
+    /// No rebalancing shall be done, as a new insight or the age of the insight shall determine whether to exit
+    /// the positions.
+    /// Rules:
+    ///    1. On Up insight, increase position size by percent
+    ///    2. On Down insight, decrease position size by percent
+    ///    3. On Flat insight, move by percent towards 0
+    ///    4. On expired insight, perform a Flat insight'''
+    /// </summary>
+    public class AccumulativeInsightPortfolioConstructionModel : PortfolioConstructionModel
+    {
+        private List<Symbol> _removedSymbols = new List<Symbol>();
+        private readonly InsightCollection _insightCollection = new InsightCollection();
+        private DateTime? _nextExpiryTime;
+        private readonly double _percent;
+        private readonly Dictionary<Insight, int> usedInsight = new Dictionary<Insight, int>();
+        private readonly Dictionary<Insight, int> expiredList = new Dictionary<Insight, int>();
+        private readonly Dictionary<Symbol, double> positionSizes = new Dictionary<Symbol, double>();
+                
+        /// <summary>
+        /// Initialize a new instance of <see cref="AccumulativeInsightPortfolioConstructionModel"/>
+        /// </summary>
+        /// <param name="percent">The amount of portfolio to allocate to a single insight</param>
```

Comment:
Doc should include the range. Since it's called  percent, users may think its in the range of [0, 100], but it appears to be within the range of [0, 1]